### PR TITLE
[NUX] Additional Changes

### LINKED ISSUE #89: NUX Updates to Merchant Onboarding
To review and improve the merchant onboarding experiences and make WooCommerce + Square integration and sync settings more intuitive. To improve the ability for the Square extension to be quickly set up when used only for payment processing and  minimise the number of steps and checkboxes that the merchants have to perform the WooCommerce + Square integration and sync, making it seamless and more intuitive. This enhancement includes the following tasks:

- [ ] Map the UX flows to set up Square more quickly when used only for payment processing
- [ ] Map the UX flows to minimize the number of steps and checkboxes that the merchants have to perform the WooCommerce + Square integration and sync
- [ ] Map the additional UX flows, as necessary, for any additional merchant onboarding recommendations
- [ ] Add e2e tests where necessary

### PR DESCRIPTION
**Pull request by faisal-alvi**
### All Submissions:

* [ ] Does your code follow the [WooCommerce Sniffs](https://github.com/woocommerce/woocommerce-sniffs/) variant of WordPress coding standards?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?
* [ ] Will this change require new documentation or changes to existing documentation?

---

### Changes proposed in this Pull Request:

This PR addresses additional changes discussed in internal UAT.

### Notes taken by @Sidsector9:

- [x] Import product modal font size fix. (DEV)
- [x] Add "Changes saved" instead of just "tick". (DEV)
- [x] Enable Sandbox container "gray" is a bit dark compared to Figma. (DEV)
- [x] Add validation when sandbox fields are empty. (DEV)
- [x] Add a message saying "connected to sandbox". (DEV)
- [x] Make sure accents match. (DEV)
- [x] Add a "back" button for payment gateway settings. (DEV)
- [x] Change "Enable this gateway" to "Enable this payment method". (DEV)
- [x] Clicking "Disconnect button" navigated to [squareup.com/login](http://squareup.com/login) (Need to look into this)
- [x] Remove old settings fields instead of hiding them with CSS. (DEV) <- removed fields but a few fields but the submit button needs to be hidden by CSS
- [x] Put the "Connection" row above the sandbox settings to avoid confusion. (UI)
- [x] Woo wants the flow (both Onboarding and Settings page) to have connection to production account first. (UI)
- [x] If there's 1 location, it should be pre-selected. (DEV).
- [x] Admin still shows "To get started, set your business location".

### Notes taken by @jeffpaul:

_Here are the notes I captured, which likely overlap with most of the items from Sid's list:_
- [x] If single biz location, set that in dropdown. 
- [x] Also adjust admin notice to not call out biz location is needed if that was set in the setup wizard.
- [x] Cleanup font sizes for import products modal (question header size, checkbox font size)
- [x] Update button final states to “<check> Changes Saved”
- [x] Sandbox mode field validation that fields have data and “<check> Connected to Sandbox”
- [x] Fix button colors to match admin theme colors
- [x] Add “back to Payments” link in payment method settings
- [x] Credit cards checkbox change to “Enable credit card payments”.  <- this is done with “Enable this payment method”
- [x] Disabling Credit cards should not disable digital wallets <- (ticket created) https://github.com/woocommerce/woocommerce-square/issues/149

_The following are the items that I pinged Evan Esposito on who will work on UI/UX updates for this concern:_
- [x] Disconnecting from Square in Settings page should not redirect to wizard, should refresh to same page with disconnected button state
- [x] Move Connect to Square row out of Configure Sandbox Settings section to a new section at the top of Connect to Square — identify copy for top of this new section

_Finally, an item not for this PT but something that SHOULD be an issue in the repo that we look to prioritize in the not-too-distant future:_
- [x] Future issue, move Digital Wallets to payments level <- (ticket created) https://github.com/woocommerce/woocommerce-square/issues/148

Also the changes from:
- [x] https://10up.slack.com/archives/C01UCQGPBDJ/p1718054633339389
- [x] https://10up.slack.com/archives/C01UCQGPBDJ/p1718305073548229 

Addition to #105.
Closes #89.

### Steps to test the changes in this Pull Request:

Test the onboarding wizard, with special attention to the changes mentioned above^.

### Changelog entry

> Add - NUX-Onboarding additional changes

### CODE REVIEW

**Comment by faisal-alvi** [comment]
@Sidsector9 I've added validation for the sandbox App ID and Access Token fields to ensure the form won't submit until both fields are filled, and an empty field gets a red border. However, the form is still being submitted even with incorrect values, and we see the "Connected to Sandbox" message. The same issue occurs on the settings page, where the sandbox fields are saved even if they're incorrect/invalid (you can type your name for example!), we still get `is_connected` being `true` in the data. Whereas it should not be connected and show some error. Can you please check and help fix this?

**Comment by faisal-alvi** [comment]
@jeffpaul All other changes have been addressed except the issue mentioned above^. I'm now waiting for Evan's input or any other feedback from UAT.

**Comment by Sidsector9** [comment]
@faisal-alvi I've fixed the onboarding sandbox connection. You can use the same logic in the Square settings connection once you get the designs. Let me know if you need any assistance.

Also I noticed the following issues:

### Console error:

Screenshot 2024-06-10 at 5 58 38 PM

### Button freeze:

Screenshot 2024-06-10 at 5 57 49 PM

Previously, the "Next" button animated (busy state) until all settings were saved, now it just freezes which can confuse the merchant.

**Comment by jeffpaul** [comment]
> Put the "Connection" row above the sandbox settings to avoid confusion. (UI) <-waiting for UI

> Woo wants the flow (both Onboarding and Settings page) to have connection to production account first. (UI) <-waiting for UI

> Move Connect to Square row out of Configure Sandbox Settings section to a new section at the top of Connect to Square — identify copy for top of this new section <-waiting for UI

Handling for all of these, I believe, should be available in the "Square Settings (Business Location, Sandbox Settings, Sync Settings, Advanced Settings)" Figma: https://www.figma.com/design/fMysPY0YQRXWWLCU0RL2En/WooCommerce-Square-Plugin-Onboarding-Wizard?node-id=2212-2566&t=gCsHeUYTSSeevjPc-0

File: assets/new-user-experience/settings/settings-app.js [hunk 1]
```diff
@@ -26,29 +26,33 @@ import {
...
 		enable_sandbox = 'no',
 		sandbox_location_id = '',
+		production_location_id = '',
 		is_connected = false,
 		connection_url = '',
 		disconnection_url = '',
 		locations = [],
 	} = settings;
 
+	console.log(locations);
```
Comment (dkotter) [inline assets/new-user-experience/settings/settings-app.js]: Assuming we can remove this

File: includes/AJAX.php [hunk 2]
```diff
@@ -165,6 +175,17 @@ public function import_products_from_square( $request = null ) {
...
 
+	/**
+	 * Update the connected_page_visited option.
+	 * This will be used to show a message to visit the onboarding
+	 * wizard and the onboarding wizard submenu until found `true`.
+	 *
+	 * @since 4.7.0
+	 */
+	public function update_connected_page_visited() {
+		update_option( 'wc_square_connected_page_visited', true );
```
Comment (dkotter) [inline includes/AJAX.php]: Should we set the autoload option on this to `false`? Or is this option always needed?

File: includes/Handlers/Products.php [hunk 3]
```diff
@@ -175,6 +177,40 @@ private function add_product_sync_hooks() {
...
+		$help_tabs = $screen->get_help_tabs();
+		if ( ! isset( $help_tabs['woocommerce_onboard_tab'] ) ) {
+			return;
+		}
+
+		$updated_help_tab = $help_tabs['woocommerce_onboard_tab'];
+
+		$square_text  = '<h2>' . __( 'Square Onboarding Setup Wizard', 'woocommerce-square' ) . '</h2>';
+		$square_text .= '<p>' . __( 'If you need to access the Square onboarding setup wizard again, please click on the button below.', 'woocommerce-square' ) . '</p>' .
+			'<p><a href="' . admin_url( 'admin.php?page=woocommerce-square-onboarding' ) . '" class="button button-primary">' . __( 'Setup wizard', 'woocommerce-square' ) . '</a></p>';
```
Comment (dkotter) [inline includes/Handlers/Products.php]:
```suggestion
		$square_text  = '<h2>' . esc_html__( 'Square Onboarding Setup Wizard', 'woocommerce-square' ) . '</h2>';
		$square_text .= '<p>' . esc_html__( 'If you need to access the Square onboarding setup wizard again, please click on the button below.', 'woocommerce-square' ) . '</p>' .
			'<p><a href="' . esc_url( admin_url( 'admin.php?page=woocommerce-square-onboarding' ) ) . '" class="button button-primary">' . esc_html__( 'Setup wizard', 'woocommerce-square' ) . '</a></p>';
```

**Review by dkotter — CHANGES_REQUESTED** [review]

File: includes/AJAX.php (hunk 2 above)
Comment (faisal-alvi) [reply includes/AJAX.php]: I think autoload for `update_option` defaults to `null`, not sure if that works or not. Though I did pass `false`.

**Review by dkotter — APPROVED** [review]

**Comment by faisal-alvi** [comment]
ESlint fixes made Today in the `wizard_page` brought a lot of conflicts here. 😿 I will fix it asap.

**Comment by faisal-alvi** [comment]
Merging this to the `wizard_page` branch (#105) so that we have only one main branch for NUX.

**Pull request merged**